integrated_channels/cornerstone/client.py

```diff
@@ -94,6 +101,17 @@ def create_course_completion(self, user_id, payload):  # pylint: disable=unused-
         )
         return response.status_code, response.text
 
+    def _convert_invalid_course_ids(self, course_id):
+        """
+        Regex check a course ID to see if it contains any invalid chars. If it does then encode the string, otherwise
+        return the original course ID.
+        """
+        re2 = re.compile(r"[|<>.&%\s\\/\“]+")
+        if re2.search(course_id):
```

Comment:
I guess I feel like, if we are converting course keys to send to CSOD, we should just convert all of the keys since it might look weird/buggy if some keys are in the traditional format and some keys are in the encoded format.  Thoughts?

Reply:
The problem is that regenerating course keys has far reaching effects since I believe those keys are used as IDs on the Cornerstone side. It is possible any dropped and recreated courses would lose linked learner data and enrollments on the Cornerstone side. At a minimum, there would suddenly be two of every course and we'd have to handle that with additional manual clean up code. This solution is for backwards compatibility. (A comment on this in code would not be amiss, @alex-sheehan-edx )

Reply:
Makes perfect sense -- I neglected to consider the fact that there are already edX courses in CSOD tenants and we would have effectively abandoned those records on the CSOD side.